crates/biome_js_analyze/src/analyzers/nursery/no_duplicate_else_if.rs

```diff
@@ -0,0 +1,216 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::{AnyJsExpression, JsIfStatement, JsLogicalOperator, JsSyntaxKind};
+use biome_rowan::{AstNode, SyntaxNodeCast};
+
+declare_rule! {
+    /// Disallow duplicate conditions in if-else-if chains
+    ///
+    /// if-else-if chains are commonly used when there is a need to execute only one branch
+    /// (or at most one branch) out of several possible branches, based on certain conditions.
+    ///
+    /// Two identical test conditions in the same chain are almost always a mistake in the code.
+    /// Unless there are side effects in the expressions,
+    /// a duplicate will evaluate to the same true or false value as the identical expression earlier in the chain,
+    /// meaning that its branch can never execute.
+    ///
+    /// Please note that this rule does not compare conditions from the chain with conditions inside statements
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// if (a) {
+    ///     foo();
+    /// } else if (b) {
+    ///     bar();
+    /// } else if (b) {
+    ///     baz();
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// if (a) {
+    ///     foo();
+    /// } else if (b) {
+    ///     bar();
+    /// } else if (c) {
+    ///     baz();
+    /// }
+    /// ```
+    ///
+    pub NoDuplicateElseIf {
+        version: "next",
+        name: "noDuplicateElseIf",
+        recommended: true,
```

Comment:
It would be nice to have source and source_kind fields
```suggestion
        recommended: true,
        source: RuleSource::Eslint("no-dupe-else-if"),
        source_kind: RuleSourceKind::Inspired // or SameLogic
```

Reply:
By default, `source_kind` is set to `RukeSourceKind::SameLogic`. Thus, if it is `SameLogic` (this is), then you don't need to set this field.